opentasks/src/main/java/org/dmfs/tasks/utils/RecentlyUsedLists.java

```diff
@@ -105,9 +118,16 @@ public static Long getRecentFromList(Context context, List<Long> allowedLists)
      */
     public static void use(Context context, Long listId)
```

Comment:
we should find out how a `null` could get into the recently used lists in the first place. In think the first step should be to change the `listId` parameter here to `long`. If, at any point, `null` is passed this will result in a `NullPointerException`.

Reply:
Do you mean to have that in the next release, so if this happens, we at least may have a stacktrace of the crash?

I spent about 30 minutes now trying to figure out how it can happen without success. Since this `ContentSet` in `EditTaskFragment` can come from multiple places it's hard to track it down. Let me know if I should spend more time with it.

I've also looked into 702994023bf6873cf72865e74098e453c20cdcdc a bit..
(That big change in `TaskListWidgetProvider` doesn't look very nice.)
How about maybe re-implementing this recently used task feature? As I understand it is only a convenience thing for the users, so if we disregard the already stored sharedprefs entry on people's devices and start to collect the data in a new prefs entry again, it wouldn't be a big problem.
Not sure if that would solve the 'null' issue though, I haven't looked into that much in that original changeset.

Reply:
Yes we should do that in the next release. This should be a `long` anyway, there is no point in allowing `null` in the recently used lists.
I don't think we should spend more time on debugging this. Let's make sure it fails fast when something is wrong. That also improves chances of discovering the bug during monkey tests.